src/NServiceBus.AcceptanceTests/Recoverability/When_exception_by_feature_is_treated_as_unrecoverable.cs

```diff
@@ -0,0 +1,102 @@
+namespace NServiceBus.AcceptanceTests.Recoverability
+{
+    using System;
+    using System.Threading.Tasks;
+    using AcceptanceTesting;
+    using AcceptanceTesting.Customization;
+    using EndpointTemplates;
+    using Features;
+    using NUnit.Framework;
+
+    public class When_exception_by_feature_is_treated_as_unrecoverable : NServiceBusAcceptanceTest
+    {
+        [Test]
+        public async Task Should_move_to_defined_error_queue()
+        {
+            var context = await Scenario.Define<Context>()
+                .WithEndpoint<EndpointWithFailingHandler>(b => b
+                    .DoNotFailOnErrorMessages()
+                    .When((session, ctx) => session.SendLocal(new InitiatingMessage
+                    {
+                        Id = ctx.TestRunId
+                    }))
+                )
+                .WithEndpoint<ErrorSpy>()
+                .Done(c => c.MessageMovedToErrorQueue)
+                .Run();
+
+            Assert.IsTrue(context.MessageMovedToErrorQueue);
+        }
+
+        class Context : ScenarioContext
+        {
+            public bool MessageMovedToErrorQueue { get; set; }
+        }
+
+        class EndpointWithFailingHandler : EndpointConfigurationBuilder
+        {
+            public EndpointWithFailingHandler()
+            {
+                EndpointSetup<DefaultServer>((config, context) =>
+                {
+                    config.SendFailedMessagesTo(Conventions.EndpointNamingConvention(typeof(ErrorSpy)));
+                });
+            }
+
+            class InitiatingHandler : IHandleMessages<InitiatingMessage>
+            {
+                public Context TestContext { get; set; }
+
+                public Task Handle(InitiatingMessage initiatingMessage, IMessageHandlerContext context)
+                {
+                    throw new CustomException();
+                }
+            }
+
+            class MyCustomFeature : Feature
+            {
+                public MyCustomFeature()
+                {
+                    Defaults(s => s.AddUnrecoverableException(typeof(SimulatedException)));
```

Comment:
How would you do it then? We cannot do it in the feature context since then we have a dependency between the recoverability feature and everything else that wants to add an exception to the exclusion list